infra/modules/notifications/data/variables.tf

```diff
@@ -0,0 +1,9 @@
+variable "name" {
+  type        = string
+  description = "Name of the notifications project/application"
+}
+
+variable "domain_name" {
+  type        = string
+  description = "The domain name to use for the email identity"
+}
```

Comment:
`data` modules should only have a single variable to make them simple. The whole point is that using the data module should be easier than using the data sources directly. Per my previous comment about the `resources` module let's get rid of `name` and keep `domain_name`.